packages/commons/src/envUtils.ts

```diff
@@ -0,0 +1,369 @@
+ *
+ * You can also provide a default value, which will be returned if the environment variable is not set instead of throwing an error.
+ *
+ * @example
+ * ```ts
+ * import { getBooleanFromEnv } from '@aws-lambda-powertools/commons/utils/env';
+ *
+ * const myEnvVar = getBooleanFromEnv({
+ *   key: 'MY_ENV_VAR',
+ *   defaultValue: true,
+ * });
+ * ```
+ *
+ * @param options - The options for getting the boolean.
+ * @param options.key - The key of the environment variable.
+ * @param options.defaultValue - The default value to return if the environment variable is not set.
+ * @param options.errorMessage - Optional error message to throw if the environment variable is not set and no default value is provided. Defaults to `"Environment variable <key> is required"`.
+ */
+const getBooleanFromEnv = ({
+  key,
+  defaultValue,
+  errorMessage,
+}: GetBooleanFromEnvOptions): boolean => {
+  const value = getStringFromEnv({
+    key,
+    defaultValue: String(defaultValue),
+    errorMessage,
+  });
+
+  const parsedValue = value.toLowerCase();
+
+  if (parsedValue !== 'true' && parsedValue !== 'false') {
+    throw new Error(`Environment variable ${key} must be a boolean`);
+  }
+
+  return parsedValue === 'true';
+};
+
+const truthyValues = ['1', 'y', 'yes', 't', 'true', 'on'];
+
+/**
+ * Get a truthy boolean from the environment variables.
+ *
+ * Truthy values are: `1`, `y`, `yes`, `t`, `true`, `on`.
+ *
+ * @example
+ * ```ts
+ * import { getTruthyBooleanFromEnv } from '@aws-lambda-powertools/commons/utils/env';
+ *
+ * const myEnvVar = getTruthyBooleanFromEnv({
+ *   key: 'MY_ENV_VAR',
+ *   errorMessage: 'MY_ENV_VAR is required for this function',
+ * });
+ * ```
+ *
+ * By default, the value is trimmed before being converted to a boolean and always required.
+ *
+ * You can also provide a default value, which will be returned if the environment variable is not set instead of throwing an error.
+ *
+ * @example
+ * ```ts
+ * import { getTruthyBooleanFromEnv } from '@aws-lambda-powertools/commons/utils/env';
+ *
+ * const myEnvVar = getTruthyBooleanFromEnv({
+ *   key: 'MY_ENV_VAR',
+ *   defaultValue: true,
+ * });
+ * ```
+ *
+ * @param options - The options for getting the truthy boolean.
+ * @param options.key - The key of the environment variable.
+ * @param options.defaultValue - The default value to return if the environment variable is not set.
+ * @param options.errorMessage - Optional error message to throw if the environment variable is not set and no default value is provided. Defaults to `"Environment variable <key> is required"`.
+ */
+const getTruthyBooleanFromEnv = ({
+  key,
+  defaultValue,
+  errorMessage,
+}: GetBooleanFromEnvOptions): boolean => {
+  const value = getStringFromEnv({
+    key,
+    defaultValue: String(defaultValue),
+    errorMessage,
+  });
+
+  return truthyValues.includes(value.toLowerCase());
+};
+
+const falsyValues = ['0', 'n', 'no', 'f', 'false', 'off'];
```

Comment:
Nit: I prefer to use a `Set` for these sort of checks.

Reply:
See this [other comment](https://github.com/aws-powertools/powertools-lambda-typescript/pull/3945#discussion_r2101964936).